src/wallet/coinselection.cpp

```diff
@@ -302,6 +302,16 @@ bool KnapsackSolver(const CAmount& nTargetValue, std::vector<OutputGroup>& group
 
 void OutputGroup::Insert(const CInputCoin& output, int depth, bool from_me, size_t ancestors, size_t descendants) {
     m_outputs.push_back(output);
+    CInputCoin& coin = m_outputs.back();
+    coin.m_fee = coin.m_input_bytes < 0 ? 0 : m_effective_feerate.GetFee(coin.m_input_bytes);
```

Comment:
I think I have asked this before, but why can `m_input_bytes` ever be below zero here? Perhaps it would be good to have a comment to explain that.

Reply:
`m_input_bytes` is initialized to `-1` to indicate that it hasn't been calculated yet.

Reply:
But this is at the point where we calculate the effective value of UTXOs, so we need to know the size. Why would we want to mitigate a missing size here rather than throwing?

Reply:
Since the intention is to not change behavior in this PR, I think I will leave this as is for now.

Additionally I don't think it is guaranteed that when we add a `CInputCoin` to an `OutputGroup` that we do know the size. It could be for a preset input or an input not in the wallet (there is a PR for that) where we add those coins to an `OutputGroup` and just don't use the effective value calculation. In those instances, the `m_input_bytes` may not be set.

Reply:
A comment would still be a good idea :)